src/uct/cuda/cuda_ipc/cuda_ipc_ep.h

```diff
@@ -1,35 +1,53 @@
 /**
-* Copyright (C) Mellanox Technologies Ltd. 2018.  ALL RIGHTS RESERVED.
-* See file LICENSE for terms.
-*/
+ * Copyright (C) Mellanox Technologies Ltd. 2018.  ALL RIGHTS RESERVED.
+ * Copyright (c) 2018, NVIDIA CORPORATION. All rights reserved.
+ * See file LICENSE for terms.
+ */
 
 #ifndef UCT_CUDA_IPC_EP_H
 #define UCT_CUDA_IPC_EP_H
 
 #include <uct/api/uct.h>
 #include <uct/base/uct_iface.h>
 #include <ucs/type/class.h>
+#include "cuda_ipc_md.h"
+
+typedef struct uct_cuda_ipc_rem_seg  uct_cuda_ipc_rem_seg_t;
+
+
+typedef struct uct_cuda_ipc_rem_seg {
+    uct_cuda_ipc_rem_seg_t *next;
+    CUipcMemHandle         ph;         /* Memory handle of GPU memory */
+    CUdeviceptr            d_bptr;     /* Allocation base address */
+    size_t                 b_len;      /* Allocation size */
+    int                    dev_num;    /* GPU Device number */
+} uct_cuda_ipc_rem_seg_t;
```

Comment:
I think caching is not part of this PR. right?

Reply:
Correct. I've missed this. I'll remove this.